### PR TITLE
py-spy 0.3.7 (new formula)

### PR DESCRIPTION
**Pull request by messense**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

https://github.com/benfred/py-spy

### CODE REVIEW

File: Formula/py-spy.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  def caveats
+    <<~EOS
+      py-spy requires root privileges so you will need to run
+      `sudo py-spy`.
+      You should be certain that you trust any software you grant root privileges.
+    EOS
+  end
+
+  test do
+    output = shell_output("#{bin}/py-spy --help 2>&1")
```
Comment (messense) [inline Formula/py-spy.rb:24]: Not sure how to do test with sudo, so l've opted to do just this.

File: Formula/py-spy.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    system "cargo", "install", *std_cargo_args
+  end
+
+  def caveats
+    <<~EOS
+      py-spy requires root privileges so you will need to run
+      `sudo py-spy`.
+      You should be certain that you trust any software you grant root privileges.
+    EOS
+  end
```
Comment (SMillerDev) [inline Formula/py-spy.rb:21]: This should be in the upstream documentation

File: Formula/py-spy.rb (hunk 1 above)
Comment (SMillerDev) [reply Formula/py-spy.rb:24]: Try running it without sudo and trying to `assert_match` the error message.

File: Formula/py-spy.rb (hunk 2 above)
Comment (messense) [reply Formula/py-spy.rb]: https://github.com/benfred/py-spy#when-do-you-need-to-run-as-sudo

**Review by nandahkrishna — APPROVED** [review]

**Pull request closed without merging**